modules/account-assignments/variables.tf

```diff
@@ -7,3 +7,9 @@ variable "account_assignments" {
     principal_type      = string
   }))
 }
+
+variable "wait_group_creation" {
+  description = "A parameter to use for data resources to depend on"
+  type        = string
```

Comment:
Instead of using a string type here, can we make it an array so that we can pass multiple dependencies instead of just one?

Reply:
That makes sense, I will test it and update the PR accordingly

Reply:
Done and it works

Reply:
I think it makes sense to have it a list of strings that simplifies the interface if there are multiple resources to depend on, while for the regular single dependency wrapping it as a list is a minor overhead. 

It could also be also a key/value map to give context to the strings but that is probably over complicating the simple situation...

I'm fine with any of the options, so your call :)

Reply:
Can a `null_resource` be avoided?

Reply:
@nitrocode this seems to be the recommended workaround when a dependency is required and the module contains data sources, see the linked articles

I'm not sure how else to implement it, if you have a suggestion I'll happily test it